.github/workflows/ci.yml

```diff
@@ -869,6 +866,15 @@ jobs:
         run: |
           conda info
           conda list
+      - name: Start local ray cluster
+        run: |
+          ray start --head --port=6379 --object-store-memory=1000000000
+          while [ $? -ne 0 ]; do
+            echo "Failed to start ray cluster. retrying..."
```

Comment:
Should we add a `sleep` here just in case?.. also I would like to make this a count-limited thing, so it stops trying at, say, 10 tries or something like that.

Unrelated - how does bash scripting work on Windows?..

Reply:
You can configure what shell the job launches, and it looks like we're choosing bash:
```
  test-clean-install-windows:
    needs: [lint-flake8, lint-black, lint-mypy, test-api, test-headers]
    runs-on: windows-latest
    defaults:
      run:
        shell: bash -l {0}
```